frontend-react/src/hooks/network/UseRequestConfig.test.ts

```diff
@@ -0,0 +1,241 @@
+import { act, renderHook } from "@testing-library/react-hooks";
+import { rest } from "msw";
+import { setupServer } from "msw/node";
+
+import {
+    createRequestConfig,
+    endpointHasMethod,
+    RSRequestConfig,
+} from "../../network/api/NewApi";
+import { MyApi, MyApiItem } from "../../network/api/test-tools/MockApi";
+
+import useRequestConfig, {
+    deletesData,
+    hasData,
+    needsData,
+    needsTrigger,
+} from "./UseRequestConfig";
+
+const mockConsoleError = jest.spyOn(global.console, "error");
+
+const handlers = [
+    /* Returns a list of two fake api items */
+    rest.get("https://test.prime.cdc.gov/api/test/test", (_req, res, ctx) => {
+        return res(
+            ctx.status(200),
+            ctx.json([new MyApiItem("1"), new MyApiItem("2")])
+        );
+    }),
+    /* Returns a single item by id */
+    rest.post<RSRequestConfig, {}, MyApiItem | any>(
+        "https://test.prime.cdc.gov/api/test/test/:id",
+        (req, res, ctx) => {
+            //@ts-ignore
+            const { id } = req.params || "0";
+            return res(ctx.status(200), ctx.json(new MyApiItem(id)));
+        }
+    ),
+
+    rest.put<RSRequestConfig, {}, MyApiItem>(
+        "https://test.prime.cdc.gov/api/test/test/:id",
+        (req, res, ctx) => {
+            //@ts-ignore
+            const { id } = req.params || "0";
+            return res(ctx.status(200), ctx.json(new MyApiItem(id)));
+        }
+    ),
+
+    rest.patch<RSRequestConfig, {}, MyApiItem>(
+        "https://test.prime.cdc.gov/api/test/test/:id",
+        (req, res, ctx) => {
+            //@ts-ignore
+            const { id } = req.params || "0";
+            return res(ctx.status(200), ctx.json(new MyApiItem(id)));
+        }
+    ),
+
+    rest.delete<RSRequestConfig, {}, MyApiItem | any>(
+        "https://test.prime.cdc.gov/api/test/test/:id",
+        (req, res, ctx) => {
+            if (!req.headers.get("authorization")?.includes("TOKEN")) {
+                return res(ctx.status(401));
+            } else {
+                return res(ctx.status(200));
+            }
+        }
+    ),
+];
+
+/* TEST SERVER TO USE IN `.test.ts` FILES */
+const testServer = setupServer(...handlers);
+
+describe("useRequestConfig", () => {
```

Comment:
would be great to see test for loading state

Reply:
Ended up moving the loading logic to useEndpoint, check that test file to see it. MUCH simpler from that layer since really it should be set to true until either data or an error message become present. That's how the effect is set up in UseEndpoint